lib/Service/Appointments/MailService.php

```diff
@@ -0,0 +1,178 @@
+
+		if($user === null) {
+			throw new ServiceException('Could not find organizer');
+		}
+
+		$fromEmail = $user->getEMailAddress();
+		$fromName = $user->getDisplayName();
+
+
+		$sys = $this->getSysEmail();
+		$message = $this->mailer->createMessage()
+			->setFrom([$sys => $fromName])
+			->setTo([$booking->getEmail() => $booking->getDisplayName()])
+			->setReplyTo([$fromEmail => $fromName]);
+
+
+		$template = $this->mailer->createEMailTemplate('calendar.confirmAppointment');
+		$template->addHeader();
+
+		//Subject
+		$subject = $this->l10n->t('Your Appointment "%s" needs confirmation', [$config->getName()]);
+		$template->setSubject($subject);
+
+		// Heading
+		$summary = $this->l10n->t("Dear %s, please confirm your booking", [$booking->getDisplayName()]);
+		$template->addHeading($summary);
+
+		// Create Booking overview
+		$this->addBulletList($template, $this->l10n, $booking, $config->getLocation());
+
+		$bookingUrl = $this->urlGenerator->linkToRouteAbsolute('calendar.booking.confirmBooking', ['token' => $booking->getToken()]);
+		$template->addBodyButton($this->l10n->t('Confirm'), $bookingUrl);
+
+		$bodyText = $this->l10n->t('This confirmation link expires in %s hours.', [(BookingService::EXPIRY /3600)] );
+		$template->addBodyText($bodyText);
+
+		$bodyText = $this->l10n->t("If you wish to cancel the appointment after all, please contact your organizer.");
+		$template->addBodyText($bodyText);
+
+		$template->addFooter();
+
+		$message->useTemplate($template);
+
+
+		try {
+			$failed = $this->mailer->send($message);
+			if (count($failed) > 0) {
+				$this->logger->warning('Mail delivery failed for some recipients.');
+				foreach($failed as $fail) {
+					$this->logger->debug('Failed to deliver email to ' . $fail);
+				}
+			}
+		} catch (\Exception $ex) {
+			$this->logger->error($ex->getMessage(), ['exception' => $ex]);
+			throw new ServiceException('Could not send mail', $ex->getCode(), $ex);
+		}
+	}
+
+	private function addBulletList(IEMailTemplate $template,
+								   IL10N $l10n,
+								   Booking $booking,
+								   ?string $location = null):void {
+
+		$template->addBodyListItem($booking->getDisplayName(), $l10n->t('Appointment:'));
+
+		$l = $this->lFactory->findGenericLanguage();
+		$relativeDateTime = $this->dateFormatter->formatDateTimeRelativeDay(
+			$booking->getStart(),
+			'long',
+			'short',
+			new \DateTimeZone($booking->getTimezone()),
+			$this->lFactory->get('calendar',$l)
+		);
+
+		$template->addBodyListItem($relativeDateTime, $l10n->t('Date:'));
+
+		if (isset($location)) {
+			$template->addBodyListItem($location, $l10n->t('Where:'));
+		}
+		if ($booking->getDescription() !== null) {
+			$template->addBodyListItem($booking->getDescription(), $l10n->t('Description:'));
+		}
+	}
+
+	/**
+	 * @return string
+	 */
+	private function getSysEmail(): string {
+		$instanceName = $this->defaults->getName();
+		return \OCP\Util::getDefaultEmailAddress($instanceName);
```

Comment:
Usually apps add something like `no-reply` in the user part. Look for uses of `getDefaultEmailAddress` in a whole Nextcloud setup.

Reply:
good idea!